### PR TITLE
Update metadata responses - Closes #1222

### LINKED ISSUE #1222: Update metadata responses
### Description

Update the responses for the following endpoints to match the stated specifications.

Endpoint: `GET /api/v3/blockchain/apps/meta/list`
```jsonc
{
  "data": [
    {
      "chainName": "Lisk",
      "chainID": "00000001",
      "networkType": "mainnet",
    },
    {
      "chainName": "Lisk",
      "chainID":"00000001",
      "networkType": "testnet",
    },
    {
      "chainName": "Lisk DEX",
      "chainID": "00000011",
      "networkType": "mainnet",
    },
    {
      "chainName": "Lisk DEX",
      "chainID":"00000003",
      "networkType": "testnet",
    },
  ],
  "meta": {
    "count": 2,
    "offset": 0,
    "total": 1000
  }
}
```

Endpoint: `GET /api/v3/blockchain/apps/meta`
```jsonc
{
  "data": [
    {
      "chainName": "Lisk",
      "chainID": "00000001",
      "title": "",
      "description": "",
      "networkType": "mainnet",
      "isDefault": true,
      "genesisURL": "https://downloads.lisk.com/lisk/mainnet/genesis_block.json.tar.gz",
      "projectPage": "https://lisk.com",
      "appPage": "",
      "logo": {
        "png": "https://downloads.lisk.com/lisk/images/logo.png",
        "svg": "https://downloads.lisk.com/lisk/images/logo.svg",
      },
      "backgroundColor": "#0981D1",
      "serviceURLs": [
        {
          "http": "https://service.lisk.com",
          "ws": "wss://service.lisk.com",
        }
      ],
      "explorers": [
        {
          "url": "https://lisk.observer",
          "txnPage": "/transactions",
        },
        {
          "url": "https://liskscan.com",
          "txnPage": "/transactions",
        },
      ],
      "appNodes": [
        {
          "url": "https://mainnet.lisk.com",
          "maintainer": "Lightcurve GmbH",
        },
        {
          "url": "wss://mainnet.lisk.com",
          "maintainer": "Lightcurve GmbH",
        },
      ],
    },
  ],
  "meta": {
    "count": 1,
    "offset": 0,
    "total": 1000
  },
  "links": {}
}
```

Endpoint: `GET /api/v3/blockchain/apps/meta/tokens`
```jsonc
{
  "data": [
    {
      "chainID": "00000001",
      "chainName": "Lisk",
      "tokenID": "00000000",
      "tokenName": "Lisk",
      "description": "Base token for the Lisk ecosystem",
      "denomUnits": [
        {
          "denom": "beddows",
          "decimal": 0,
        },
        {
          "denom": "lsk",
          "decimal": 8,
          "aliases": ["Lisk"], // Optional
        },
      ],
      "baseDenom": "beddows",
      "displayDenom": "lisk",
      "symbol": "LSK",
      "logo": {
        "png": "https://downloads.lisk.com/lisk/images/tokens/lsk.png",
        "svg": "https://downloads.lisk.com/lisk/images/tokens/lsk.svg",
      },
    },
  ],
  "meta": {
    "count": 1,
    "offset": 0,
    "total": 1000
  },
  "links": {}
}
```

Event: `metadata.change'
```jsonc
{
  mainnet: ['Lisk'],
  testnet: ['Kalipo'],
  betanet: ['Lisk', 'Lisk Dex'],
}
```

### Acceptance Criteria

- The responses are updated as specified above
- Integration tests are up-to-date
- The swagger specification reflects the above changes

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #1222 

### How was it solved?

- [x] Update gateway:
  - [x] Response mappings
  - [x] Params mapping
- [x] Update `app-registry` service
  - [x] Controller 
  - [x] implementation to retrieve metadata information
- [x] Integration tests are up-to-date
- [x] The swagger specification reflects the above changes
- [x] Update event payload for `update.metadata` 

### How was it tested?
Local

### CODE REVIEW

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/gateway/sources/version3/mappings/blockchainAppMetadata.js [hunk 1]
```diff
@@ -14,22 +14,24 @@
...
+		http: '=,string',
+		ws: '=,string',
 	}],
-	explorers: '=',
 	logo: {
 		png: '=,string',
 		svg: '=,string',
 	},
+	explorers: '=',
+	appNodes: '=',
```
Comment (sameersubudhi) [inline services/gateway/sources/version3/mappings/blockchainAppMetadata.js:35]: Can we also explicitly declare the structure for the `explorers` and `appNodes` properties like we did for `serviceURLs` here?

File: services/gateway/sources/version3/mappings/blockchainAppTokenMetadata.js [hunk 2]
```diff
@@ -16,13 +16,16 @@
...
+	networkType: 'network,string',
+	description: '=,string',
+	logo: {
+		png: '=,string',
+		svg: '=,string',
+	},
+	symbol: '=,string',
+	displayDenom: '=,string',
+	baseDenom: '=,string',
+	denomUnits: '=',
```
Comment (sameersubudhi) [inline services/gateway/sources/version3/mappings/blockchainAppTokenMetadata.js:30]: Same as above.

File: services/blockchain-app-registry/shared/metadata.js [hunk 3]
```diff
@@ -78,33 +55,31 @@ const getBlockchainAppsMetaList = async (params) => {
...
-			.concat(formatResponseEntries(nonDefaultApps));
+		blockchainAppsMetaList.data = defaultApps.concat(nonDefaultApps);
 	}
 
 	blockchainAppsMetaList.data = blockchainAppsMetaList.data
 		.slice(params.offset, params.offset + params.limit);
 
-	const [{ count: total }] = await applicationsDB.rawQuery('SELECT COUNT(DISTINCT(name)) as count from applications');
+	// TODO: Use count method directly once support for custom column-based count added https://github.com/LiskHQ/lisk-service/issues/1188
+	const [{ count: total }] = await applicationsDB.rawQuery('SELECT COUNT(chainName) as count from applications');
```
Comment (sameersubudhi) [inline services/blockchain-app-registry/shared/metadata.js:82]:
```suggestion
	const [{ total }] = await applicationsDB.rawQuery('SELECT COUNT(chainName) as total from applications');
```

File: services/blockchain-app-registry/shared/metadata.js [hunk 4]
```diff
@@ -140,28 +111,39 @@ const getBlockchainAppsMetadata = async (params) => {
...
+			const chainMeta = JSON.parse(chainMetaString);
+			return chainMeta;
 		},
-		{ concurrency: response.length },
+		{ concurrency: blockchainAppsMetadata.data.length },
 	);
 
-	const total = await applicationsDB.count(params);
+	// TODO: Use count method directly once support for custom column-based count added https://github.com/LiskHQ/lisk-service/issues/1188
+	const [{ count: total }] = await applicationsDB.rawQuery('SELECT COUNT(chainName) as count from applications');
```
Comment (sameersubudhi) [inline services/blockchain-app-registry/shared/metadata.js:146]:
```suggestion
	const [{ total }] = await applicationsDB.rawQuery('SELECT COUNT(chainName) as total from applications');
```

File: services/blockchain-app-registry/shared/metadata.js [hunk 5]
```diff
@@ -199,14 +181,25 @@ const getBlockchainAppsTokenMetadata = async (params) => {
...
+			return parsedTokenMeta;
+		},
+		{ concurrency: tokensResultSet.length },
 	);
 
-	blockchainAppsTokenMetadata.data = formatResponseEntriesForTokens(response);
-
-	const total = await tokensDB.count(params);
+	// TODO: Use count method directly once support for custom column-based count added https://github.com/LiskHQ/lisk-service/issues/1188
+	const [{ count: total }] = await applicationsDB.rawQuery('SELECT COUNT(tokenName) as count from tokens');
```
Comment (sameersubudhi) [inline services/blockchain-app-registry/shared/metadata.js:202]:
```suggestion
	const [{ total }] = await applicationsDB.rawQuery('SELECT COUNT(tokenName) as total from tokens');
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/blockchain-app-registry/shared/metadata.js [hunk 6]
```diff
@@ -199,14 +181,25 @@ const getBlockchainAppsTokenMetadata = async (params) => {
...
-	const response = await tokensDB.find(
-		params,
-		Object.getOwnPropertyNames(tokensIndexSchema.schema),
+	const tokensResultSet = await tokensDB.find(params, ['network', 'chainID']);
+
+	blockchainAppsTokenMetadata.data = await BluebirdPromise.map(
+		tokensResultSet,
+		async (tokenMeta) => {
+			const [{ appDirName }] = await applicationsDB.find(
+				{ network: tokenMeta.network, chainID: 1 },
```
Comment (sameersubudhi) [inline services/blockchain-app-registry/shared/metadata.js:190]: Why do we have the `chainID` hardcoded here?

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

**Review by priojeetpriyom — APPROVED** [review]

**Review by priojeetpriyom — APPROVED** [review]

**Pull request merged**